File: leaoai_python_refatorado_descompactado/app/models/Factory/projectmeasureitem.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.core.database import Base
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .projectblock import ProjectBlock

class ProjectMeasureItem(Base):
    __tablename__ = "ProjectMeasureItems"

    Id = Column(Integer, primary_key=True, index=True)
    ProjectBlockId = Column(Integer, ForeignKey('projectblocks.Id'))
    Name = Column(String)
    Width = Column(Float, nullable=True)
    Height = Column(Float, nullable=True)
    Length = Column(Float, nullable=True)
    Weight = Column(Float, nullable=True)
    NameEditable = Column(Boolean)

    # Define relationships here
    ProjectBlock = relationship('ProjectBlock')

File: Framework.Data/Models/Factory/ProjectMeasureItem.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Framework.Data.Models.Factory
{
	// PROJECT MEASURE ITEM
	public class ProjectMeasureItem
	{
		[Key]
		public int Id { get; set; }

		[ForeignKey("ProjectBlock")]
		public int ProjectBlockId { get; set; }
		public virtual ProjectBlock ProjectBlock { get; set; }

		[Required]
		public string Name { get; set; }

		public string Width { get; set; }
		public string Height { get; set; }
		public string Length { get; set; }
		public string Weight { get; set; }

		public bool NameEditable { get; set; } = false;
	}
}
